lib/feature/contact_support/widgets/contact_support/contact_support_sheet.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:app/feature/contact_support/contact_support.dart';
+import 'package:app/generated/generated.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:ui_components_lib/ui_components_lib.dart';
+
+/// Helper function to show [ContactSupportSheet].
+Future<void> showContactSupportSheet({
+  required BuildContext context,
+  required ContactSupportMode mode,
+  bool showHiding = true,
+}) {
+  final title = switch (mode) {
+    ContactSupportMode.initiatedByUser =>
+      LocaleKeys.contactSupportBottomSheetHeaderInitiatedByUser.tr(),
+    ContactSupportMode.initiatedByCrash =>
+      LocaleKeys.contactSupportBottomSheetHeaderInitiatedByCrash.tr(),
+  };
+
+  final subtitle = switch (mode) {
+    ContactSupportMode.initiatedByUser =>
+      LocaleKeys.contactSupportBottomSheetTextInitiatedByUser.tr(),
+    ContactSupportMode.initiatedByCrash =>
+      LocaleKeys.contactSupportBottomSheetTextInitiatedByCrash.tr(),
+  };
+
+  return showCommonBottomSheet(
+    context: context,
+    title: title,
+    subtitle: subtitle,
+    body: (_, __) => BlocProvider(
+      create: (context) => ContactSupportBloc(),
+      child: BlocListener<ContactSupportBloc, ContactSupportState>(
+        listener: (context, state) {
+          state.when(
+            initial: () {
```

Comment:
Is this state avoid during first launch? Imho better use special state like `empty` or `close` to make pop action.
Also you can use `whenOrNull` to avoid listing all of events

Reply:
It's called when changed to initial from any other, but not when the block constructor was called. The declarative approach forces us to use triggers when transitioning to a state, rather than creating additional states for triggers. Also, using when() instead of whenOrNull() is just for exhaustiveness. But anyway, I've changed it already :)